xtb/qcschema/harness.py

```diff
@@ -0,0 +1,86 @@
+# This file is part of xtb.
+#
+# Copyright (C) 2020 Sebastian Ehlert
+#
+# xtb is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# xtb is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with xtb.  If not, see <https://www.gnu.org/licenses/>.
+"""Integration with the QCArchive infrastructure"""
+
+from .. import __version__
+from ..interface import Calculator, Param, XTBException, VERBOSITY_MUTED
+import qcelemental as qcel
+
+
+_methods = {
+    "GFN2-xTB": Param.GFN2xTB,
+    "GFN1-xTB": Param.GFN1xTB,
+    "GFN0-xTB": Param.GFN0xTB,
+    "GFN-FF": Param.GFNFF,
+}
+
+
+def run_qcschema(input_data: qcel.models.AtomicInput) -> qcel.models.AtomicResult:
```

Comment:
may want to allow `input_data` to be dict so that your users don't have to know about qcel

Reply:
Not sure if I want to get not validated input here, convincing qcel to produce garbage input to force an error in the API took me a little while.